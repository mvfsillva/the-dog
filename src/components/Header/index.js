import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Title from 'components/Title'
import If from 'components/common/If'

const HeaderStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-transform: ${props => (props.uppercase ? 'uppercase' : 'capitalize')};
  color: ${props => props.theme.colors.gray.default};
`

const Header = ({ id, className, headline, subtitle, ...props }) => (
  <HeaderStyled id={id} className={className} {...props}>
    <Title size="h1">{headline}</Title>
    <If test={subtitle}>
      <Title size="h2">{subtitle}</Title>
    </If>
  </HeaderStyled>
)

Header.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  headline: PropTypes.node.isRequired,
  subtitle: PropTypes.node,
  uppercase: PropTypes.bool,
}

Header.defaultProps = {
  id: null,
  className: null,
  subtitle: null,
  uppercase: false,
}

export { HeaderStyled }
export default Header
