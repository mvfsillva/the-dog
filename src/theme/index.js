const baselineGrid = 4
const interfaceGrid = 8

const breakpoints = ['576px', '768px', '992px', '1200px']

const colors = {
  primary: {
    default: '#6800c3',
    fallback: '#a55cff',
    text: '#fff',
  },
}

export default {
  baselineGrid,
  interfaceGrid,
  breakpoints,
  colors,
}
