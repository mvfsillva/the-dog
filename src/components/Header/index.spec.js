import React from 'react'
import 'jest-styled-components'
import Header from '.'
import theme from '../../theme'
import { shallowWithTheme, mountWithTheme } from '../../../config/test/enzyme.theme'

const shallow = shallowWithTheme(theme)
const mount = mountWithTheme(theme)

const wrap = (props = {}) => shallow(<Header headline="headline" {...props} />).dive()

describe('[Component: Header]', () => {
  it('mounts with different combination props', () => {
    mount(<Header headline="headline" />)
    mount(<Header headline="headline" subtitle="subtitle" />)
    mount(<Header headline="headline" uppercase />)
    mount(<Header headline="headline" hero />)
    mount(<Header headline="headline" subtitle="subtitle" uppercase hero />)
  })

  it('renders headline when passed in', () => {
    const wrapper = wrap({ headline: 'test' })
    expect(wrapper.contains('test')).toBe(true)
  })

  it('renders subtitle when passed in', () => {
    const wrapper = wrap({ subtitle: 'test' })
    expect(wrapper.contains('test')).toBe(true)
  })

  it('renders props when passed in', () => {
    const wrapper = wrap({ id: 'foo' })
    expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
  })
})
