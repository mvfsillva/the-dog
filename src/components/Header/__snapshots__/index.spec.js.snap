// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Header] matches snapshot 1`] = `
<styled.div
  className={null}
  hero={false}
  id={null}
  uppercase={false}
>
  <Title
    className="thedog"
    color="grayscale"
    size="h1"
  />
  <If
    test={null}
  >
    <Title
      color="grayscale"
      size="h2"
    />
  </If>
</styled.div>
`;
